apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceState.java

```diff
@@ -94,7 +100,11 @@ public void addTextHeader(String headerValue) {
                     if (0 <= value && value <= 1.0) {
                         // ensure proper rounding of sample rate to minimize storage
                         // even if configuration should not allow this, any upstream value might require rounding
-                        double rounded = Math.round(value * 10000d) / 10000d;
+                        double rounded = Math.round(value * PRECISION) / PRECISION;
+                        if (value > 0 && rounded == 0) {
+                            // avoid rounding to zero
+                            rounded = 1d / PRECISION;
+                        }
 
```

Comment:
Looks like a duplication. Can you use `RoundedDoubleConverter` to do that, preferably using the same hard coded precision used in `CoreConfiguration` for the `sample_rate` config?

Reply:
Good point, fixed in the last commit.